### PR TITLE
Removed dependency on version manifest

### LINKED ISSUE #255: clusterConfig failed to validate
A trial deploy of the operator on OpenShift 4.4 in combination with deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml configured for mongo 3.6 appeared to be successful, with all pods in running status, but mongod is not operational. I looked in the example-openshift-mongodb-0 pod's log, and mongodb-agent is repeatedly logging the following error:

```
[2020-11-24T13:14:31.584-00] [.error] [cm/main/cm.go:mainLoop:456] [13:14:31.584] Error loading desired cluster configs : [13:14:31.584] Error retrieving cluster config from '/var/lib/automation/config/cluster-config.json' : [13:14:31.584] Error validating syntax of cluster config : clusterConfig failed to validate:
&{builds 0xc000299020} : Invalid type. Expected: array, given: null
```

A cat of /var/lib/automation/config/cluster-config.json, as generated by the operator, does show a null indeed:

```
  ...
  "mongoDbVersions": [
    {
      "name": "3.6",
      "builds": null
    }
  ...
```

Any idea what could be causing this problem?
My mongodb.com_v1_mongodb_openshift_cr.yaml  looks like this:

```
apiVersion: mongodb.com/v1
kind: MongoDB
metadata:
  name: example-openshift-mongodb
spec:
  members: 3
  type: ReplicaSet
  version: "3.6"
  featureCompatibilityVersion: "3.6"
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: my-user
      db: admin
      passwordSecretRef:
        name: my-user-password
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
      scramCredentialsSecretName: my-scram
  statefulSet:
    spec:
      template:
        spec:
          containers:
            - name: "mongodb-agent"
              env:
                - name: MANAGED_SECURITY_CONTEXT
                  value: "true"
            - name: "mongod"
              env:
                - name: MANAGED_SECURITY_CONTEXT
                  value: "true"
```

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [N/A] Have you opened an Issue before filing this PR?
* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [X] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

This PR upgrades the agent version to a version which supports allowing versions of MongoDB which are not present in the version manifest.

* The version manifest has been removed from the operator image.
* A dummy version config is being used in the automation config.
* The agent version has been upgraded.

closes #255

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,7 +31,7 @@ BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 BUNDLE_IMG ?= controller-bundle:$(VERSION)
 
 # Image URL to use all building/pushing image targets
-IMG ?= quay.io/mongodb/mongodb-kubernetes-operator:0.5.0 # replace with localhost:5000/mongodb-kubernetes-operator locally
```
Comment (chatton) [inline Makefile:34]: removing the official registry as default to prevent any accidental pushes

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -32,7 +32,7 @@ spec:
             - name: OPERATOR_NAME
               value: "mongodb-kubernetes-operator"
             - name: AGENT_IMAGE # The MongoDB Agent the operator will deploy to manage MongoDB deployments
-              value: quay.io/mongodb/mongodb-agent:10.19.0.6562-1
+              value: quay.io/mongodb/mongodb-agent:10.27.0.6772-1
```
Comment (chatton) [inline config/manager/manager.yaml:35]: this version of the operator allows us to provide the dummy configuration

File: Makefile (hunk 1 above)
Comment (bznein) [reply Makefile:34]: 🥳🥳🥳

File: config/manager/manager.yaml (hunk 2 above)
Comment (bznein) [reply config/manager/manager.yaml:35]: 💯

**Review by bznein — APPROVED** [review]
Negative diff, short PR that makes a huge impact on the operator? LGTM!

**Pull request merged**